core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalWriteConverterCtx.scala

```diff
@@ -31,11 +31,14 @@ class ExternalWriteConverterCtx(nodeDesc: NodeDesc, totalNumOfRows: Int) extends
 
   /**
     * This method closes the communication after the chunks have been closed
+    * @throws ExternalFrameConfirmationException in case of confirmation failure.
+    *         This has also effect of Spark stopping the current job and
+    *         rescheduling it
     */
   override def closeChunks(): Unit = {
     try{
-      externalFrameWriter.waitUntilAllWritten()
-    }finally {
+      externalFrameWriter.waitUntilAllWritten(10)
```

Comment:
Secods, but it is a still a draft. I need to make it configurable, but didn't have enough time at PyData